src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -111,9 +111,13 @@ void CompositionalMultiphaseBase::registerDataOnMesh( Group & meshBodies )
   ConstitutiveManager const & cm = domain.getConstitutiveManager();
 
   // 1. Set key dimensions of the problem
-  MultiFluidBase const & fluid0 = cm.getConstitutiveRelation< MultiFluidBase >( m_fluidModelNames[0] );
-  m_numPhases     = fluid0.numFluidPhases();
-  m_numComponents = fluid0.numFluidComponents();
+  // Empty check needed to avoid accessing m_fluidModelNames when running in schema generation mode.
+  if( !m_fluidModelNames.empty() )
+  {
+    MultiFluidBase const & fluid0 = cm.getConstitutiveRelation< MultiFluidBase >( m_fluidModelNames[0] );
+    m_numPhases = fluid0.numFluidPhases();
+    m_numComponents = fluid0.numFluidComponents();
+  }
```

Comment:
There are actually more checks that happen a bit later in the simulation (in `CompositionalMultiphaseBase::initializePreSubGroups()`) that should catch an empty fluid model list. This one here only helps us avoid the crash when in schema generation mode, which never proceeds to solve anything.

Reply:
I would argue that `CompositionalMultiphaseBase` is responsible for both `solverStep`, `applyBoundaryConditions` and schema generation. That's a lot for a single class. 😉 

We have a very strong link from our `xml` files to implementation.
Sometimes it could be interesting to split responsibilities (when there is an issue, an IT will always propose to add a layer. 🧅 )
Having an extra layer that actually builds/fills all our events, outputs, solvers... could be an interesting move. (Solvers would be there to solve only: `CompositionalMultiphaseBase` currently depends on `xml`).

Having a broader view, we could enhance our consistency checks, we could maybe support multiple version of input files (allowing our users smoother transitions)...
When you mention `CommandLineOptions` this is part of the idea I think (obviously, it would not be raw `CommandLineOptions` as is, but something one step further).

However, great for the additional checks, not need to further think about it then.